component/client/chat/ClientIndependentChatRoom.tsx

```diff
@@ -22,27 +23,24 @@ class ClientIndependentChatRoom extends React.Component<any, IClientIndependentC
     super();
     this.state = {
       currentUrl: Constants.NOOP_URL,
+      userID: Constants.NOOP_ID,
       username: Constants.NOOP_USERNAME
     };
-    storage.get("username")
-      .then((username) => {
-        if (username) {
+    storage.get(Constants.STORAGE_KEY_USERNAME).then((username) => {
+      storage.get(Constants.STORAGE_KEY_ID).then((userID) => {
```

Comment:
Even if React does not optimize it, render() would be called twice. That would not affect performance. If one is slower than the other, why not show the faster one first?
I don't think your way would make error show up any differently. Promise are executed asynchronously, so that the component continues to mount before the storage data gets back.